Add recipe and helper for deploying media

jonte opened the pull request:
This adds a recipe and inc file for deploying test media. Media is
extracted from a tarball specified by SRC_URI, and known media types are
extracted and grouped in to packages, installed in /usr/share.

Audio filed can be installed by including test-multimedia-audio, video
files are in test-multimedia-video, etc.

jonte commented:
Looking for feedback on this, what do you think? The .inc-file should allow users to supply their own test media, and the default gstreamer test media is hosted by Pelagicore and free for anyone to use.

rpannek commented on recipes-multimedia/test-multimedia/gstreamer-test-multimedia.bb:
```diff
@@ -0,0 +1,12 @@
+
+include test-multimedia.inc
+
+HOMEPAGE = "https://139.162.170.90/public/"
```
It's super weird to have a hard coded IP address in here, once this is down we'll not be able to point to something new via DNS. Even worse, when I open this URL Firefox warns me about a insecure certificate:

![music-data](https://cloud.githubusercontent.com/assets/8579375/20795224/441f45be-b7d0-11e6-9ee1-a9baaab0a4d2.png)

rpannek commented on recipes-multimedia/test-multimedia/gstreamer-test-multimedia.bb:
```diff
@@ -0,0 +1,12 @@
+
+HOMEPAGE = "https://139.162.170.90/public/"
+LICENSE = "CC-BY-NC-1.0 | CC-BY-SA-1.0 | CC-BY-1.0"
+SRC_URI = "https://139.162.170.90/public/media.tar.gz;name=mediatgz"
```
Same here.

It would also be nice to have some information about where this media is comming from originally.

jonte replied:
Yeah. I agree. We don't have a name connected to this server though. Maybe we should look in to investing in that. If we decide we want this type of functionality, we can buy a name, and get a proper cert through letsencrypt.

jonte commented:
There were some functional issues with the previous patch:
- Packages were created for file types with no files present
- "meta" packages (called file packs in the code) for video, audio and picture extensions were generated even when those extensions did not have any files present

I fixed the above issues, and also fixed the URI issue pointed out by @rpannek. The cert issue is still present, and is being worked on, but does not affect the code in the patch.

I removed the WIP status, and I consider this ready for merging, so please re-review :)

jonte commented on recipes-multimedia/test-multimedia/test-multimedia.inc:
```diff
@@ -0,0 +1,104 @@
+
+S = "${WORKDIR}"
+
+python () {
```
Yes.. So line 38-43 will probably cause issues. I have tried to provoke errors and cause this to break, but I have not succeeded. I will put 38-43 in a function and call it from do_install and populate_packages_prepend instead. Thanks for spotting this!

jonte replied:
I fixed this in the latest update.

jonte commented on recipes-multimedia/test-multimedia/gstreamer-test-multimedia.bb:
```diff
@@ -0,0 +1,12 @@
+
+HOMEPAGE = "https://testmedia.pelagicore.net/public/"
+LICENSE = "CC-BY-NC-1.0 | CC-BY-SA-1.0 | CC-BY-1.0"
+SRC_URI = "https://testmedia.pelagicore.net/public/media.tar.gz;name=mediatgz"
```
No structure is required, and I will add a note about this to the .inc-file.

jeena commented on recipes-multimedia/test-multimedia/test-multimedia.inc:
```diff
@@ -0,0 +1,104 @@
+
+    # Remove filetype with no files present
+    for ftype in ftypes:
+        files = bb.utils.find_all_with_ext(d.expand("${WORKDIR}"), ftype)
```
This might be nitpicking but a file name extension is not the same as a file type.

jonte commented:
Final fiddle from my part.. I moved the inc file to a bbclass. This makes it easier to add the "test media" capability to a recipe while retaining existing functionality. I added this as a separate commit to make it easier to read. When accepting this, please squash the commits together, or let me know and I will do it.
